Makefile

```diff
@@ -54,45 +59,75 @@ export DOCKER_COMPOSE_SHARED_HEADER
 
 all: init dockerbuild
 
-init:	initbuild initdirs initcomposevars
+init: initbuild initdirs
 	@echo "using ${DOCKER_COMPOSE} as compose command"
 
-initbuild:
-ifeq (,$(wildcard ./Dockerfile))
+initbuild: Dockerfile .env docker-compose.yml migrid-httpd-init.sh
+	@echo "initialised environment for build"
+
+Dockerfile:
 	@echo
 	@echo "*** No Dockerfile selected - defaulting to rocky9 ***"
 	@echo
 	ln -s Dockerfile.rocky9 Dockerfile
 	@sleep 2
-endif
-ifeq (,$(wildcard ./.env))
+
+.env:
 	@echo
 	@echo "*** No deployment environment selected - defaulting to development ***"
 	@echo
 	ln -s development.env .env
 	@sleep 2
-endif
-ifeq (,$(wildcard ./docker-compose.yml))
+
+docker-compose.yml:
 	@echo
 	@echo "*** No docker-compose.yml selected - defaulting to development ***"
 	@echo
 	ln -s docker-compose_development.yml docker-compose.yml
 	@sleep 2
-endif
+
+migrid-httpd-init:
 	sed 's@#unset @unset @g;s@#export @export @g' migrid-httpd.env > migrid-httpd-init.sh
 
-initdirs:
+initdirs: initcomposevars
 	mkdir -p external-certificates
 	mkdir -p certs
 	mkdir -p httpd
 	mkdir -p mig
 	mkdir -p state
-	mkdir -p log/migrid
-	mkdir -p log/migrid-io
-	mkdir -p log/migrid-openid
-	mkdir -p log/migrid-sftp
-	mkdir -p log/migrid-webdavs
-	mkdir -p log/migrid-ftps
+	mkdir -p ${VOLATILE_ROOT}/mig_system_run
+	mkdir -p ${VOLATILE_ROOT}/openid_store
+	mkdir -p ${PERSISTENT_ROOT}/freeze_home
+	mkdir -p ${PERSISTENT_ROOT}/mrsl_files
```

Comment:
This "mrsl_files" name seem to conflict a bit with the `docker-compose_production_bind.yml` mrsl_home volume:

  mrsl_home:
    # Volume used to contain the migrid mrsl_home
    driver: local
    driver_opts:
      type: none
      device: ${PERSISTENT_ROOT}/mrsl_home
      o: bind

So either will have to adjust to the other:)

Reply:
Thanks, fixed along with the missing `sharelink_home` volume declaration raised below. 
Added the `checkconf` self-test in the `migrid` container of the development docker compose files.